spec.md

```diff
@@ -546,6 +548,10 @@ The tags MUST be in lexical order.
 
 When using the `last` query parameter, the `n` parameter is OPTIONAL.
 
+The registry SHOULD support at least `n=1000` tags in a single response.
+The client SHOULD avoid requesting more than `n=1000` tags in a single request.
+If the registry is unable to return all tags requested by the client, the registry MUST return a `413 Content Too Large` with `TAG_LIMIT_EXCEEDED` error <sup>[code-15](#error-codes)</sup> and include the header `OCI-Tag-Limit: <limit>`, where `<limit>` is the maximum tags per request supported by the registry.
```

Comment:
```suggestion
If the registry is unable to return all tags requested by the client, the registry MUST return a `413 Content Too Large` with `TAG_LIMIT_EXCEEDED` error <sup>[code-15](#error-codes)</sup> and include the header `OCI-Tag-Limit: <limit>`, where `<limit>` is the maximum tags supported by the registry for the request.
```

Reply:
I'm not following the reason for this change. Is this saying the registry would give a different limit for different requests? So if a client tried to pull 1000 tags, the registry might respond that the limit is 500. And then when the client asks for 500 tags, the registry would be permitted to return an error that says the limit is now 100 tags? Or is this saying that other kinds of requests would have different limits? Perhaps an example would help.

Reply:
I'm saying I don't see a reason to force all registries to adhere to this rule that they MUST have only one maximum limit for all requests to get tags.

There could be a variance by repository {local/remote, public/private, ...} or another type of variance like total size of response.

Reply:
That seems like it would add complication on both clients and registries. Are registries doing this today? If not, it would be useful for clients to be able to reuse a previously seen limit from a registry to avoid additional round trips.

Reply:
Adding this error will break existing correct (IMO) clients that do the right (IMO) thing by following Link headers when present so that we can loudly break any clients that are silently incorrect because they don't follow a Link header. This doesn't seem like a good tradeoff to me.